### PR TITLE
registry-replacer: add flag -maxConcurrency

### PR DESCRIPTION
**Pull request by hongkailiu**
Follow up https://coreos.slack.com/archives/CHY2E1BL4/p1613657069247400

The periodic-prow-auto-config-brancher job is flaky.
https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-prow-auto-config-brancher/1362401502118809600#1:build-log.txt%3A3642

The job run a batch of ci-tools. This tool `registry-replacer` is added the last and it uses concurrency to handle ci-operator's config.

This should improve the success rate.

/cc @openshift/openshift-team-developer-productivity-test-platform 

/hold

Will run a job with the built image.

### CODE REVIEW

**Review by alvaroaleman — APPROVED** [review]

File: cmd/registry-replacer/main.go [hunk 1]
```diff
@@ -93,6 +98,10 @@ func main() {
 	if err != nil {
 		logrus.WithError(err).Fatal("failed to gather options")
 	}
+	if opts.maxConcurrency == 0 {
+		opts.maxConcurrency = runtime.GOMAXPROCS(0)
```
Comment (alvaroaleman) [inline cmd/registry-replacer/main.go:102]: Can we use a fixed higher number like 500, this tool needs to make over 6k http requests currently?

Comment (alvaroaleman) [reply cmd/registry-replacer/main.go:102]: 6000 / 4 would be 1.5k seconds, assuming each request takes one second and we never have to retry

File: cmd/registry-replacer/main.go [hunk 2]
```diff
@@ -57,6 +60,7 @@ func gatherOptions() (*options, error) {
 	flag.BoolVar(&o.selfApprove, "self-approve", false, "If the bot should self-approve its PR.")
 	flag.BoolVar(&o.ensureCorrectPromotionDockerfile, "ensure-correct-promotion-dockerfile", false, "If Dockerfiles used for promotion should get updated to match whats in the ocp-build-data repo")
 	flag.Var(o.ensureCorrectPromotionDockerfileIngoredRepos, "ensure-correct-promotion-dockerfile-ignored-repos", "Repos that are being ignored when ensuring the correct promotion dockerfile in org/repo notation. Can be passed multiple times.")
+	flag.IntVar(&o.maxConcurrency, "concurrency", 0, "Maximum number of concurrent in-flight goroutines to handle files.")
```
Comment (alvaroaleman) [inline cmd/registry-replacer/main.go:63]: Just make it 500 here instead of defaulting later :)

Comment (hongkailiu) [reply cmd/registry-replacer/main.go]: Updated.

**Review by alvaroaleman — APPROVED** [review]

**Comment by hongkailiu** [comment]
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_ci-tools/1700/pull-ci-openshift-ci-tools-master-images/1362485944443211776

https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-prow-auto-config-brancher/1362498114627309568#1:build-log.txt%3A4
is using the image from ^

Seems working.

/hold cancel

**Pull request merged**